File: EmpProject/src/cn/yh/vo/Emp.java
package cn.yh.vo;

import java.io.Serializable;
import java.util.Date;

@SuppressWarnings("serial")
public class Emp implements Serializable {
	private Integer empno;
	private String ename;
	private String job;
	private Date hiredate;
	private double sal;
	private double comm;
	private String photo;
	private String note;
	private Emp mgr;
	private Dept dept;
	public Emp(){}
	public Emp(Integer empno, String ename, String job, Date hiredate, double sal, double comm, String photo,
			String note, Emp mgr, Dept dept) {
		super();
		this.empno = empno;
		this.ename = ename;
		this.job = job;
		this.hiredate = hiredate;
		this.sal = sal;
		this.comm = comm;
		this.photo = photo;
		this.note = note;
		this.mgr = mgr;
		this.dept = dept;
	}

	public Integer getEmpno() {
		return empno;
	}
	public void setEmpno(Integer empno) {
		this.empno = empno;
	}
	public String getEname() {
		return ename;
	}
	public void setEname(String ename) {
		this.ename = ename;
	}
	public String getJob() {
		return job;
	}
	public void setJob(String job) {
		this.job = job;
	}
	public Date getHiredate() {
		return hiredate;
	}
	public void setHiredate(Date hiredate) {
		this.hiredate = hiredate;
	}
	public double getSal() {
		return sal;
	}
	public void setSal(double sal) {
		this.sal = sal;
	}
	public double getComm() {
		return comm;
	}
	public void setComm(double comm) {
		this.comm = comm;
	}
	public void setMgr(Emp mgr) {
		this.mgr = mgr;
	}
	public Emp getMgr() {
		return mgr;
	}
	public void setDept(Dept dept) {
		this.dept = dept;
	}
	public Dept getDept() {
		return dept;
	}
	public String getPhoto() {
		return photo;
	}
	public void setPhoto(String photo) {
		this.photo = photo;
	}
	public String getNote() {
		return note;
	}
	public void setNote(String note) {
		this.note = note;
	}
	

}

File: EmpProject/WebRoot/js/emp.js
function validateEmpno(){
	return validateEmpty("empno",/^\d{4}$/);
}
function validateEname(){
	return validateEmpty("ename");
}
function validateJob(){
	return validateEmpty("job");
}
function validateHiredate(){
	return validateEmpty("hiredate",/^\d{4}-\d{2}-\d{2}$/);
}
function validateSal(){
	return validateEmpty("sal",/^\d+(\.\d{1,2})?$/);
}
function validateComm(){
	return validateEmpty("comm",/^\d+(\.\d{1,2})?$/);
}

function validateinsert(){
	return validateEmpno() && validateEname() && validateJob() && validateHiredate() && validateSal() && validateComm();
}
function validateupdate(){
	return validateEname() && validateJob() && validateHiredate() && validateSal() && validateComm();
}